pandas/core/groupby/groupby.py

```diff
@@ -1319,6 +1328,16 @@ def f(self, **kwargs):
                 except Exception:
                     result = self.aggregate(
                         lambda x: npfunc(x, axis=self.axis))
+
+                    # coerce the columns if we can
+                    if isinstance(result, DataFrame):
```

Comment:
Shouldn't this logic be in `_try_cast`? OK as a follow up just want to confirm

Reply:
no, _try_cast is for 1d results, maybe I can make this cleaner

Reply:
I sligthly refactored